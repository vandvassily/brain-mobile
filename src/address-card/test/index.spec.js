import AddressCard from '..';
import { mount } from '../../../test';

const addressInfo = {
  address: '北京市东城区',
  name: 'vassily',
  phone: '13000000000'
};

test('create a AddressCard', () => {
  const wrapper = mount(AddressCard);

  expect(wrapper).toMatchSnapshot();
});

test('create a AddressCard with props', () => {
  const wrapper = mount(AddressCard, {
    propsData: {
      addressInfo
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('valid area placeholder confirm', async () => {
  const wrapper = mount(AddressCard, {
    propsData: {
      addressInfo
    }
  });

  const field = wrapper.findAll('.address-card').at(0);
  field.trigger('click');

  expect(wrapper.emitted('click')).toBeTruthy();
});
