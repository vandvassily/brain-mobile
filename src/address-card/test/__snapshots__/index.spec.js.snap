// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a AddressCard 1`] = `
<div class="address-card">
  <div class="card-info"><i class="van-icon van-icon-location-o">
      <!---->
    </i> <span></span> <span></span></div>
  <div class="address-info"><span></span></div>
  <div class="edit-icon"><i class="van-icon van-icon-edit">
      <!---->
    </i></div>
</div>
`;

exports[`create a AddressCard with props 1`] = `
<div class="address-card">
  <div class="card-info"><i class="van-icon van-icon-location-o">
      <!---->
    </i> <span>vassily</span> <span>13000000000</span></div>
  <div class="address-info"><span>北京市东城区</span></div>
  <div class="edit-icon"><i class="van-icon van-icon-edit">
      <!---->
    </i></div>
</div>
`;
